code/modules/bitrunning/server/map_handling.dm

```diff
@@ -84,12 +84,32 @@
 	for(var/datum/lazy_template/virtual_domain/available in SSbitrunning.all_domains)
 		if(map_key == available.key && points >= available.cost)
 			generated_domain = available
-			RegisterSignal(generated_domain, COMSIG_LAZY_TEMPLATE_LOADED, PROC_REF(on_template_loaded))
-			generated_domain.lazy_load()
-			return TRUE
 
-	return FALSE
+	if(!generated_domain)
+		return FALSE
+
+	var/list/mob/lucky_ghosts
+	if(generated_domain.mission_min_candidates)
+		playsound(src, 'sound/machines/chime.ogg', 50, TRUE)
+		say("Loading advanced NPCs...")
+		var/list/mob/candidates = SSpolling.poll_ghost_candidates("Do you want to play as a virtual [generated_domain.spawner_role] in a bitrunner domain?", ROLE_GHOST_ROLE, ROLE_GHOST_ROLE, 15 SECONDS, POLL_IGNORE_SHUTTLE_DENIZENS, TRUE)
+		for(var/amount in 1 to generated_domain.mission_max_candidates)
+			LAZYADD(lucky_ghosts, pick_n_take(candidates))
+
+		if(length(lucky_ghosts) < generated_domain.mission_min_candidates)
+			notify_ghosts("Not enough candidates for [generated_domain.spawner_role]! Aborting mission!")
+			playsound(src, "sound/machines/buzz-[pick("sigh", "two")].ogg", 50, TRUE)
+			say("Error! Unable to load advanced NPCs. Please try again or select different domain.")
```

Comment:
Is this spammable (can a bitrunner span deadchat over and over again)?

Reply:
fixed